x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/search/aggregations/GeoLineAggregatorTests.java

```diff
@@ -431,9 +432,11 @@ public void testGeoLine_Terms_TSDB_simplified() throws IOException {
     private void assertGeoLine(SortOrder sortOrder, String group, InternalGeoLine geoLine, TestTSAssertionResults tsx, boolean complete) {
         long[] expectedAggPoints = tsx.expectedAggPoints.get(group);
         double[] expectedAggSortValues = tsx.expectedAggSortValues.get(group);
-        String prefix = "GeoLine[sort=" + sortOrder + ", use-timestamps=" + tsx.useTimestampField + "]";
+        String prefix = "GeoLine[sort=" + sortOrder + ", use-timestamps=" + tsx.useTimeSeriesAggregation + ", group='" + group + "']";
         assertThat(prefix + " is complete", geoLine.isComplete(), is(complete));
-        assertThat(prefix + " contents", geoLine.line(), isGeoLine(tsx.useTimestampField, expectedAggPoints));
+        // old geo_line has a bug whereby it can produce lines with truncation happening in the middle instead of the end
+        int checkCount = tsx.useTimeSeriesAggregation ? expectedAggPoints.length : min(expectedAggPoints.length / 2, geoLine.line().length);
```

Comment:
This is the workaround. Since the bug causes truncation to happen in the middle of the line instead of the end, we just need to test the correct end of the line. Since the documents with missing points are always 1/3 of the documents (see TestData class), we can be sure the first 2/3 of the line is good. So we test the first half only. We need to do this for both the points and the timestamps, since they are correlated and the issue will come through on both.